### PR TITLE
fix issue #656: add 'modules' keyword arg to load Redis extension modules

### LINKED ISSUE #656: no way to pass `--loadmodule` argument to command line
I would like to be able to start redis in tests, with redis-server extended with Redis extension modules.

The `--loadmodule` command line argument is here for this. Unfortunately it is not supported by `pytest-redis`.

I tried other possibilities, but actually it is not possible:
- to pass any keyword argument ( see issue #297)
- to specify a configuration file for the redis server started with `pytest-redis`
- regarding modules, without allowing it in a configuration file, it is forbidden to use the `MODULE LOAD` command once server is started

So, the easiest would really be to add `modules` as a valid keyword argument to the the Redis executor.

### PR DESCRIPTION
**Pull request by mguijarr**
Here is a pull request to close #656 , taking into account comments in : https://github.com/ClearcodeHQ/pytest-redis/pull/657

(sorry had to create a new pull request... Not super at ease with github vs gitlab !)

### CODE REVIEW

File: newsfragments/656.misc.rst
Comment (fizyk) [inline newsfragments/656.misc.rst:1]: Please change it to 656.feature.rst it's worth highlighting in the changelog :)

**Review by fizyk — APPROVED** [review]

**Comment by fizyk** [comment]
@mguijarr one last thing.... would you be able to add some tests as well?

**Comment by fizyk** [comment]
Oh, one last thing, please mention in the documentation that it will not work for noproc fixtures.

File: newsfragments/656.misc.rst
Comment (mguijarr) [reply newsfragments/656.misc.rst:1]: done

**Comment by mguijarr** [comment]
> @mguijarr one last thing.... would you be able to add some tests as well?

I added a test, just for `modules` keyword argument (didn't find the tests for pytest.ini and pytest command line args)

**Comment by mguijarr** [comment]
> Oh, one last thing, please mention in the documentation that it will not work for noproc fixtures.

I thought it was done already ? In the table, it has `-` for `noproc-fixture` column

File: tests/test_executor.py [hunk 1]
```diff
@@ -218,3 +218,26 @@ def test_noopredis_handles_timeout_when_waiting() -> None:
...
+        databases=4,
+        redis_timeout=config["timeout"],
+        loglevel=config["loglevel"],
+        port=redis_port,
+        host=config["host"],
+        startup_timeout=30,
+        datadir=tmpdir,
+        modules=["nonexistent.so", "nonexistent2.so"]
+    )
+    with pytest.raises(ProcessExitedWithError, match="--loadmodule nonexistent.so --loadmodule nonexistent2.so"):
```
Comment (fizyk) [inline tests/test_executor.py:240]: is there a way to load an existing module and check in tests that it's loaded? 
The ProcessExitedWithError is rather general, so any other problem starting the redis in this tests will make it pass.

Comment (mguijarr) [reply tests/test_executor.py:240]: Unfortunately there is no default module to load (to my knowledge). I am going to see if I can get a better output message from output...

**Comment by mguijarr** [comment]
> > I thought it was done already ? In the table, it has - for noproc-fixture column
> 
> Yeah, the column lists configuration for the fixture if you'd want to use the fixture factory, and pass your own values there. There's nothing else stating that if you configure from command line or config file, it will not affect the redis connected to through noproc fixture.

I just added more descriptive documentation

**Review by fizyk — APPROVED** [review]

File: pytest_redis/executor/process.py [hunk 2]
```diff
@@ -58,6 +58,7 @@ def __init__(
         syslog_enabled: bool = False,
         appendonly: str = "no",
         datadir: Optional[Path] = None,
+        modules: Optional[list[str]] = None,
```
Comment (fizyk) [inline pytest_redis/executor/process.py:61]: For python 3.8 you'd have to use List from typing.

Comment (mguijarr) [reply pytest_redis/executor/process.py]: good catch ! sorry

I fixed it

**Comment by fizyk** [comment]
@mguijarr now, there are some issues with mypy now and the ruff linter left :)

**Comment by mguijarr** [comment]
Sorry for the long delay, have been busy - ok I tried to make mypy happy and I applied ruff

**Comment by mguijarr** [comment]
Ok thanks for approval to let tests pass.. Again it fails with mypy 😭 

No idea what is wrong

**Comment by mguijarr** [comment]
I don't use mypy maybe you can make the fix yourself ? If you don't mind

**Comment by fizyk** [comment]
I'll try to take a look sometimes next week :)

File: pytest_redis/factories/proc.py [hunk 3]
```diff
@@ -92,6 +94,13 @@ def redis_proc_fixture(
         else:
             redis_datadir = tmp_path_factory.mktemp(f"pytest-redis-{request.fixturename}")
 
+        if modules:
+            redis_modules = modules
+        elif config.get("modules"):
+            redis_modules = config["modules"].split(",")
```
Comment (fizyk) [inline pytest_redis/factories/proc.py:100]: @mguijarr the problem in this line is that having the `elif config.get("modules"):` in the line above does not tell mypy, that `config["modules"]` holds any data. Could be something to address within mypy, though looking at the config dict, the "modules" is always set, changing the condition in line 99 to `elif config["modules"]:` makes mypy understand, that if it gets into line 100, the value under modules is always string, and is not possible to be None.

Comment (mguijarr) [reply pytest_redis/factories/proc.py:100]: Ok! I see. Thanks a lot for fixing this!

**Comment by mguijarr** [comment]
Looks nice 🥳  Is it ready to merge soon?

**Review by fizyk — APPROVED** [review]

**Comment by mguijarr** [comment]
Thanks a lot! 👍🏻 And a great thank you for pytest-redis, very useful

**Comment by fizyk** [comment]
Thank you for your contribution :)

Wait a bit, till I add python 3.12 proper support, and redis 7.2 into tests and I'll be releasing new version :)

**Pull request merged**